src/IssueViz/IssueVisualizationControl/ViewModels/IssueVisualizationViewModel.cs

```diff
@@ -0,0 +1,207 @@
+            selectionEvents.SelectedFlowChanged += SelectionEventsOnSelectedFlowChanged;
+            selectionEvents.SelectedLocationChanged += SelectionEvents_SelectedLocationChanged;
+        }
+
+        public string Description => currentIssue?.Issue?.Message;
+
+        public string RuleKey => currentIssue?.Issue?.RuleKey;
+
+        public string RuleHelpLink => string.IsNullOrEmpty(RuleKey) ? null : ruleHelpLinkProvider.GetHelpLink(RuleKey);
+
+        public AnalysisIssueSeverity Severity => currentIssue?.Issue?.Severity ?? AnalysisIssueSeverity.Info;
+
+        public IAnalysisIssueVisualization CurrentIssue
+        {
+            get => currentIssue;
+            private set
+            {
+                currentIssue = value;
+                OnPropertyChanged(null);
+            }
+        }
+
+        public IAnalysisIssueFlowVisualization CurrentFlow
+        {
+            get => currentFlow;
+            set
+            {
+                currentFlow = value;
+
+                if (isBindingUpdatedOutsideOfControl)
+                {
+                    OnPropertyChanged(nameof(CurrentFlow));
+
+                    LocationListItems = BuildLocationListItems(currentFlow);
+                    OnPropertyChanged(nameof(LocationListItems));
+                }
+                else
+                {
+                    selectionEvents.SelectedFlow = currentFlow;
+                }
+            }
+        }
+
+        public IReadOnlyList<ILocationListItem> LocationListItems { get; private set; }
+
+        public LocationListItem CurrentLocationListItem
+        {
+            get => currentLocation;
+            set
+            {
+                currentLocation = value;
+
+                if (isBindingUpdatedOutsideOfControl)
+                {
+                    OnPropertyChanged(nameof(CurrentLocationListItem));
+                }
+                else
+                {
+                    selectionEvents.SelectedLocation = currentLocation?.Location;
+                }
+            }
+        }
+
+        private void SelectionEvents_SelectedIssueChanged(object sender, IssueChangedEventArgs e)
+        {
+            CurrentIssue = e.Issue;
+        }
+
+        private void SelectionEventsOnSelectedFlowChanged(object sender, FlowChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            CurrentFlow = e.Flow;
+
+            isBindingUpdatedOutsideOfControl = false;
+        }
+
+        private void SelectionEvents_SelectedLocationChanged(object sender, LocationChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            if (e.Location == null)
+            {
+                CurrentLocationListItem = null;
+            }
+            else
+            {
+                var selectedLocationListItem = LocationListItems?
+                    .OfType<LocationListItem>()
+                    .FirstOrDefault(x => x.Location == e.Location);
```

Comment:
This looks questionable: it seems odd that we have to walk the list comparing the items to work out which is the current location.
We might need to revisit this if we want also want to render the current location adornment differently.

Reply:
Maybe dictionary then? <int stepNumber, LocationListItem>

Reply:
I'd leave it for now. The problem for me is that it's the control that is working it out, rather than it being a property on the IAnalysisIssueLocationVisualization.